crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -771,6 +771,24 @@ impl Timeline {
 
         self.room().send_multiple_receipts(receipts).await
     }
+
+    /// Mark the room as read by sending a read receipt on the latest event, be
+    /// it visible or not.
+    ///
+    /// Returns a boolean indicating if it sent the request or not.
+    #[instrument(skip(self), fields(room_id = ?self.room().room_id()))]
+    pub async fn mark_as_read(
+        &self,
+        receipt_type: ReceiptType,
+        thread: ReceiptThread,
+    ) -> Result<bool> {
+        if let Some(event_id) = self.inner.latest_event_id().await {
+            self.send_single_receipt(receipt_type, thread, event_id).await
+        } else {
+            trace!("can't mark room as read because there's no latest event id");
```

Comment:
Should this be an error instead?

Reply:
Not sure. A timeline with no events would be an actual bug, right? But I think it could be happen transiently, if the timeline has been cleared because of a gappy sync, for instance.

Reply:
I don't think it would be a bug? Can't we have an empty room as well?

Reply:
I suppose that a minimal room would have some state events in the timeline for the room creation itself, and the members joining. The timeline doesn't filter them out, so they *should* be there, at least.

Reply:
I can't find the issue/comment, but turns out that an `m.room.create` event is not the first event you receive for a new room, it doesn't even have to be in the sync response that gives you the room for the first time.

That's why the `create` field in `RoomInfo` is optional :upside_down_face:.

Though ok, at least something should be there.

Reply:
TIL, thanks (also: oh wow).